src/ui/Startup.cs

```diff
@@ -43,10 +46,67 @@ public void ConfigureServices(IServiceCollection services)
             {
                 options.DefaultSignInScheme = CookieAuthenticationDefaults.AuthenticationScheme;
                 options.DefaultAuthenticateScheme = CookieAuthenticationDefaults.AuthenticationScheme;
-                options.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;
+                options.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;                
+                
             }).AddCookie(options =>
             {
-                options.ExpireTimeSpan = TimeSpan.FromMinutes(20);
+                options.ExpireTimeSpan = TimeSpan.FromMinutes(120);
+                options.Events = new CookieAuthenticationEvents
+                {
+                    // this event is fired everytime the cookie has been validated by the cookie middleware,
+                    // so basically during every authenticated request
+                    // the decryption of the cookie has already happened so we have access to the user claims
+                    // and cookie properties - expiration, etc..
+                    OnValidatePrincipal = async x =>
```

Comment:
Quoting references would be handy for others trying to understand this later. Url in comment would be fine.

Reply:
done